eno-core/src/test/java/fr/insee/eno/core/processing/out/steps/lunatic/LunaticInsertCodeFiltersTest.java

```diff
@@ -0,0 +1,70 @@
+package fr.insee.eno.core.processing.out.steps.lunatic;
+
+import fr.insee.eno.core.PoguesToEno;
+import fr.insee.eno.core.exceptions.business.PoguesDeserializationException;
+import fr.insee.eno.core.mappers.LunaticMapper;
+import fr.insee.eno.core.model.EnoQuestionnaire;
+import fr.insee.eno.core.parameter.EnoParameters;
+import fr.insee.eno.core.serialize.PoguesDeserializer;
+import fr.insee.lunatic.model.flat.CheckboxGroup;
+import fr.insee.lunatic.model.flat.Radio;
+import fr.insee.pogues.model.Questionnaire;
+import org.junit.jupiter.api.Test;
+
+import java.net.URISyntaxException;
+import java.net.URL;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertNotNull;
+
+class LunaticInsertCodeFiltersTest {
+
+    @Test
+    void deserialize_largeQuestionnaire() throws URISyntaxException, PoguesDeserializationException {
+        //
+        URL testPoguesFileUrl = this.getClass().getClassLoader().getResource(
+                "functional/pogues/codes-filtered/pogues-m8hgkyw0.json");
+        assert testPoguesFileUrl != null;
+        //
+        Questionnaire poguesQuestionnaire = PoguesDeserializer.deserialize(testPoguesFileUrl);
+        //
+        assertNotNull(poguesQuestionnaire);
+        assertEquals("m8hgkyw0", poguesQuestionnaire.getId());
+
+        EnoQuestionnaire enoQuestionnaire = PoguesToEno.fromObject(poguesQuestionnaire).
+                transform(EnoParameters.of(EnoParameters.Context.DEFAULT, EnoParameters.ModeParameter.CAWI));
+
+        fr.insee.lunatic.model.flat.Questionnaire lunaticQuestionnaire = new fr.insee.lunatic.model.flat.Questionnaire();
+        LunaticMapper lunaticMapper = new LunaticMapper();
+        lunaticMapper.mapQuestionnaire(enoQuestionnaire, lunaticQuestionnaire);
+
+        //
+        new LunaticInsertCodeFilters(enoQuestionnaire).apply(lunaticQuestionnaire);
+
+        //
+        Radio radio = lunaticQuestionnaire.getComponents().stream()
+                .filter(Radio.class::isInstance)
+                .map(Radio.class::cast)
+                .findAny().orElse(null);
+        assertNotNull(radio);
+        assertEquals("m8hgctb4", radio.getId());
+        assertEquals("nvl(AGE, 0) > 18", radio.getOptions().stream()
+                .filter(c -> "3".equals(c.getValue()))
+                .toList()
+                .getFirst()
+                .getConditionFilter().getValue());
+        //
```

Comment:
Pour les tests tu peux utiliser la lib assertj qui est vraiment sympa et `puissant.`
```kts
dependencies {
    testImplementation("org.assertj:assertj-core:4.0.0-M1")
}
```

Ce qui donne
```suggestion
        Assertions.assertThat(radio)
                .isNotNull()
                .extracting(ComponentType::getId)
                .isEqualTo("m8hgkyw0");
        
        Assertions.assertThat(radio.getOptions())
                .filteredOn(c -> "3".equals(c.getValue()))
                .first()
                .extracting(option -> option.getConditionFilter().getValue())
                .isEqualTo("nvl(AGE, 0) > 18");
```

là c'est juste un exemple mais sur des gros asserts un peu chian à fair tu peux faire des trucs cool en chainant et en utilisant pleins de méthodes utiles.